lib/plugin/instance.rb

```diff
@@ -576,6 +576,21 @@ def on(event_name, &block)
     DiscourseEvent.on(event_name) { |*args, **kwargs| block.call(*args, **kwargs) if enabled? }
   end
 
+  # A proxy to `DiscourseEvent.on(:site_setting_changed)` triggered when the plugin enabled setting specified by
+  # `enabled_site_setting` value is changed, including when the plugin is turned off.
+  #
+  # It is useful when the plugin needs to perform tasks like properly clearing caches when enabled/disabled
```

Comment:
```suggestion
  # A proxy to `DiscourseEvent.on(:site_setting_changed)` triggered when the plugin enabled setting specified by
  # `enabled_site_setting` value is changed, including when the plugin is turned off.
  #
  # It is useful when the plugin needs to perform tasks like properly clearing caches when enabled/disabled
  # note it will not be triggered when a plugin is installed/uninstalled by adding/removing its code
```